cmd/goal/p2pid.go

```diff
@@ -47,12 +47,12 @@ var p2pID = &cobra.Command{
 			peerKey, err := p2p.GetPrivKey(config.Local{P2PPersistPeerID: true}, dataDir)
 			if err != nil {
 				fmt.Fprintf(os.Stderr, "Error obtaining private key: %v\n", err)
-				os.Exit(1)
+				return
 			}
 			peerID, err := peer.IDFromPublicKey(peerKey.GetPublic())
 			if err != nil {
 				fmt.Fprintf(os.Stderr, "Error obtaining peerID from a key: %v\n", err)
-				os.Exit(1)
+				return
```

Comment:
Perhaps adopt the `anyError` pattern that other commands use to keep the `os.Exit(1)` case? https://github.com/algorand/go-algorand/blob/578684e28266a0ea47c356c811fd689c51070bf3/cmd/algocfg/resetCommand.go#L48

Reply:
added